node/src/components/chain_synchronizer/operations.rs

```diff
@@ -1245,6 +1285,7 @@ async fn fetch_blocks_and_state_since_genesis(ctx: &ChainSyncContext<'_>) -> Res
 async fn fetch_block_worker(
     worker_id: usize,
     latest_height_requested: Arc<AtomicU64>,
+    validator_weights: BTreeMap<PublicKey, U512>,
```

Comment:
What I would do is in this method, instead of just getting the `block_hash` from storage for `next_block_height`, I would also look up what's the validator set that should sign the block.

Reply:
This is implemented in https://github.com/casper-network/casper-node/commit/b1172ac330b0abd8ef94e017ec280fe856d022b6 by leveraging the `get_switch_block_header_at_era_id_from_storage`